x-pack/test/encrypted_saved_objects_api_integration/fixtures/api_consumer_plugin/server/index.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup, PluginInitializer } from '../../../../../../src/core/server';
+import { deepFreeze } from '../../../../../../src/core/utils';
+import {
+  EncryptedSavedObjectsPluginSetup,
+  EncryptedSavedObjectsPluginStart,
+} from '../../../../../plugins/encrypted_saved_objects/server';
+import { SpacesPluginSetup } from '../../../../../plugins/spaces/server';
+
+const SAVED_OBJECT_WITH_SECRET_TYPE = 'saved-object-with-secret';
+
+interface PluginsSetup {
+  encryptedSavedObjects: EncryptedSavedObjectsPluginSetup;
+  spaces: SpacesPluginSetup;
+}
+
+interface PluginsStart {
+  encryptedSavedObjects: EncryptedSavedObjectsPluginStart;
+  spaces: never;
+}
+
+export const plugin: PluginInitializer<void, void, PluginsSetup, PluginsStart> = () => ({
+  setup(core: CoreSetup<PluginsStart>, deps) {
+    core.savedObjects.registerType({
+      name: SAVED_OBJECT_WITH_SECRET_TYPE,
+      hidden: false,
+      namespaceAgnostic: false,
+      mappings: deepFreeze({
+        properties: {
+          publicProperty: { type: 'keyword' },
+          publicPropertyExcludedFromAAD: { type: 'keyword' },
+          privateProperty: { type: 'binary' },
+        },
+      }),
+    });
+
+    deps.encryptedSavedObjects.registerType({
+      type: SAVED_OBJECT_WITH_SECRET_TYPE,
+      attributesToEncrypt: new Set(['privateProperty']),
+      attributesToExcludeFromAAD: new Set(['publicPropertyExcludedFromAAD']),
+    });
+
+    core.http.createRouter().get(
+      {
+        path: '/api/saved_objects/get-decrypted-as-internal-user/{id}',
+        validate: { params: value => ({ value }) },
```

Comment:
__note:__ custom validation to unblock this PR as it's not important for the test plugins anyway (it seems we're hitting the bug https://github.com/elastic/kibana/issues/61652).